### ISSUE TITLE
Severity.error 42883: function strftime(unknown, timestamp with time zone) does not exist

### ISSUE THREAD

**Issue by ivanesi** [simolus3/moor#3263]
I'm trying use drift with postgres for server app. And requests to db throw error:

```
Unhandled exception:
Severity.error 42883: function strftime(unknown, timestamp with time zone) does not exist hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

The code:

```dart
Future<void> main() {

final dbConnection = PgDatabase(
    endpoint: Endpoint(
      host: 'localhost',
      database: 'postgres',
      username: 'postgres',
      password: 'postgres',
    ),
    settings: const ConnectionSettings(
      sslMode: SslMode.disable, 
    ),
  );

  final database = AppDatabase(dbConnection);
  
 print(await database.appUser.all().get());

}

@DriftDatabase(tables: [AppUser])
class AppDatabase extends _$AppDatabase {
  AppDatabase(super.e);

  @override
  int get schemaVersion => 1;
}

class AppUser extends Table {
  Int64Column get id => int64().autoIncrement()();
  DateTimeColumn get creationTime => dateTime().withDefault(currentDateAndTime)();
}
```

**Comment by ivanesi** [comment]
postgres log:


```
2024-10-04 12:57:57.992 UTC [294] ERROR:  function strftime(unknown, timestamp with time zone) does not exist at character 123
2024-10-04 12:57:57.992 UTC [294] HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2024-10-04 12:57:57.992 UTC [294] STATEMENT:  CREATE TABLE IF NOT EXISTS "app_user" ("id" bigserial PRIMARY KEY NOT NULL, "creation_time" bigint NOT NULL DEFAULT (CAST(strftime('%s', CURRENT_TIMESTAMP) AS bigint)));
```

**Comment by simolus3** [comment]
This is a known limitation - drift's datetime APIs were designed with SQLite in mind only, and won't support anything else without major changes. Regrettably, these limitations are not properly documented - I've addressed that in ea2411a686be1bc6cb7e8a28c44986c23710b595.

Basically, you'll want to use something like this instead:

```dart
class AppUser extends Table {
  IntColumn get id => integer().autoIncrement()(); // using int64 makes no sense on postgres, you're not running on the web
  Column<PgDateTime> get creationTime => customType(PgTypes.dateTime).withDefault(now())();
}
```

**Comment by ivanesi** [comment]
Thanks.  The getter 'dateTime' isn't defined for the type 'PgTypes'. So I use `PgTypes.timestampWithTimezone`.
